bin/transform-genbank

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python3
+"""
+Parse the GenBank JSON load into a metadata tsv and a FASTA file.
+"""
+import argparse
+import sys
+from pathlib import Path
+import re
+import pandas as pd
+from typing import List
+
+sys.path.insert(0, str(Path(__file__).parent.parent / "lib"))
+from utils.transform import (
+    METADATA_COLUMNS,
+    standardize_dataframe,
+    fill_default_geo_metadata,
+    write_fasta_file,
+)
+
+assert 'sequence' not in METADATA_COLUMNS, "Sequences should not appear in metadata!"
+
+def preprocess(genbank_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Rename and standardize column types and drop records where the
+    sequence length is less than 15kkb.
+    Returns the modified DataFrame.
+    """
+    mapper = {
+        'collected': 'date',
+        'submitted': 'date_submitted',
+    }
+
+    # Standardize column names, dtypes and drop entries with length less than 15kb
+    return standardize_dataframe(genbank_data, mapper)
+
+
+def standardize_strain_names(genbank_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Attempt to standardize strain names by removing extra prefixes,
+    stripping spaces, converting country code to country name, and correcting
+    known common error patterns.
+    """
+    # Compile list of regex to be used for strain name standardization
+    # Order is important here! Keep the known prefixes first!
+    regex_replacement = [
+        (r'(^SAR[S]{0,1}[-\s]CoV[-]{0,1}2/|^2019[-\s]nCoV[-_\s/]|^BetaCoV/|^nCoV-|^hCoV-19/)',''),
+        (r'(human/|homo sapien/)',''),
+        (r'^USA-', 'USA/'),
+        (r'^USACT-', 'USA/CT-'),
+        (r'^USAWA-', 'USA/WA-'),
+        (r'^HKG.', 'HongKong/'),
+    ]
+    # Add country codes to regex_replacement
+    country_codes = pd.read_csv(base / 'source-data/country-codes.tsv', header=None, sep='\t', comment="#")
+    # Remove accents and non-alphabet characters from country names
+    country_codes[1] = country_codes[1] \
+        .str.normalize("NFD") \
+        .str.encode("ascii", errors="ignore") \
+        .str.decode("utf-8") \
+        .str.replace(r'[^a-zA-Z]', '')
+    country_codes.apply(lambda c: regex_replacement.extend([
+        (rf'^{re.escape(c[0])}/', f'{c[1]}/'),
+        (rf'/{re.escape(c[0])}/', f'/{c[1]}/')]), axis=1)
```

Comment:
I looked at this again because the forcing things into ASCII caught my eye. In general, things shouldn't be forced into ASCII. But then when re-reading the surrounding context, I'm not really sure why we'd replace ISO country codes with mangled names in these strain names? Country codes seem very appropriate for use in strain names, so I think we could just leave them as-is.

Reply:
Hmm, I think it was weird that some strain names contain the full country name while others are just country codes. If it's not weird to have a mixture of both, then I'm fine with leaving the country codes as is.

Reply:
Nod. I think fine to leave them. Strain names as submitted will continually use non-standard nomenclature in various ways. If we did want to standardize geolocality in strain names, I think using the ISO codes would be more appropriate than country names anyway.